[DX] [TwigBundle] Enhance the new exception page design

sustmi opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

- [x] Fix the problem with wrapping wide lines (https://github.com/symfony/symfony/pull/20951#issuecomment-292201983)

I got motivated by recent PR https://github.com/symfony/symfony/pull/20951 and redesigned the exception page even more.

Compare before: ![before](https://cloud.githubusercontent.com/assets/885946/25052220/0756b74e-2151-11e7-98b6-c99fd9eaabec.png)
with after: ![after](https://cloud.githubusercontent.com/assets/885946/25052225/0c76581a-2151-11e7-96ff-eb502ee8e97b.png)
(Ignore the the "sf" toolbar icon in the middle of the page. This is how Firefox does full-page screenshots.)

The most noticeable changes:
- removed double line spacing (it just does not feel natural)
- file context increased from 3 to 10 lines (it helps me to better orientate in the code)
- added horizontal scrollbar for the cases when the code is wider
- the highlighted line color is more saturated in order to be noticed easily

sustmi committed:
increase context in file excerpt from 3 to 10 lines

sustmi committed:
remove padding of selected lines in order to make height of all lines equal

sustmi committed:
move code excerpt out of toggling div in order to not close when clicking into code

sustmi committed:
set borders for code excerpt to 10px on both sides

sustmi committed:
increase selected line color saturation

sustmi committed:
enable horizontal scrollbar in file excerpt

It was necessary to remove table layout and replace it with DIVs and SPANs
because it was messing with width calculations.

sustmi commented:
Wait, I found that the problem with wide lines described here: https://github.com/symfony/symfony/pull/20951#issuecomment-292201983 still happens.

sustmi committed:
do not wrap long lines

sustmi commented:
Fixed in the new commit.

robfrawley commented:
Personally, I preferred the less dense "double line" of the original.

sustmi commented:
@robfrawley, do you think that with single line spacing the code is harder to readable?
I guess that single line spacing is the most common setting in IDEs and editors (I have never seen anybody use double line spacing for PHP) and therefore I consider single line spacing more natural for reading PHP code (it creates less cognitive load).

curry684 commented:
> file context increased from 3 to 10 lines (it helps me to better orientate in the code)

👎 on that one as it then dominates the average developer's screen completely. And you can just middle/right click the filename above the snippet to see the full scrollable context if you really need it.

Totally down with bringing the line spacing down though, just tried it and yes it reads more like actual code though. And it does give room to increase the context to 5 without sacrificing call stack.

sustmi committed:
lower context in file excerpt to 5 lines

sustmi commented:
Ok, I lowered the file excerpt line context to 5. You can see the difference here:

10 lines | 5 lines
--- | ---
![10-line-context](https://cloud.githubusercontent.com/assets/885946/25064233/8a6a0b5e-21f6-11e7-97a1-373b87efa408.png) | ![5-line-context](https://cloud.githubusercontent.com/assets/885946/25064234/9199f880-21f6-11e7-9d8a-e594a70f4953.png)

What do you think?

sustmi commented:
Rebased to the current master to rerun Travis CI.

javiereguiluz commented:
I'm sorry I didn't review this on time for 3.3.0 ... I even thought this was merged! There are some things that I'm not sure about ... but others that I like. Tomorrow I'll add more details. Thanks!

apfelbox commented:
About the issues with the line count in the code excerpts… how about the way Github does this?

![2017-06-01 at 10 54](https://cloud.githubusercontent.com/assets/1032411/26672237/b9b0c3c6-46b8-11e7-905c-5072ae02b5c8.png)

Or would this be overkill?

stof commented:
@apfelbox IMO, this is overkill. You have access to the full file anyway if you want.

stof commented:
@sustmi can you rebase this on top of the Symfony 3.4 branch to fix conflicts ?

ogizanagi commented:
As I said in https://github.com/symfony/symfony/pull/22971#issuecomment-305913262, I'd love to see this merged in 3.3 as a bug fix rather than in 3.4. There is no DX nor feature to me, just UI/UX improvements and fixes for minor bu annoying behaviors. :)

nicolas-grekas commented:
@sustmi would you mind rebasing please?

curry684 commented:
I second the 3.3.1 vote. It's not a feature, just an improvement on 3.3-introduced behavior. DX/UX improvements have always been allowed in patch releases, no reason to postpone it.

sustmi commented:
I rebased the PR to 3.4 as @stof adviced. Is that all right?

sustmi commented:
I just noticed the discussion in https://github.com/symfony/symfony/pull/22971#issuecomment-305913262 so I guess you actually wanted me to rebase it to 3.3.
So I rebased it to 3.3 and I also removed commit `remove double line spacing to make it more compact and similar to IDEs` as @ogizanagi already changed line-spacing from 10px to 5px in https://github.com/symfony/symfony/pull/22971 and @nicolas-grekas seemed to like it in https://github.com/symfony/symfony/pull/22971#issuecomment-305876918 as it is consistent with GitHub look.

fabpot commented:
You also need to change the target to 3.3.

sustmi commented:
Oh, thanks for the notice. Done.

fabpot commented:
@javiereguiluz Is it ready to merge or do you need more time to review?

javiereguiluz reviewed (approved):
👍   I've reviewed everything again and it looks good to me.

In your screenshots some things look different ... for example some border are not visible:

![border-exception](https://user-images.githubusercontent.com/73419/27899468-a1a5814e-622a-11e7-9c9b-c5048ac5bdb5.png)

 ... but I've reviewed the CSS styles and they look OK, so there must be some problem with the screenshot.

fabpot commented:
Thank you @sustmi.
